packages/next-on-pages/templates/_worker.js/routes-matcher.ts

```diff
@@ -537,11 +537,12 @@ export class RoutesMatcher {
 			}
 		}
 
-		// In the `hit` phase or for external urls/redirects, return the match.
+		// In the `hit` phase or for external urls/redirects/middleware responses, return the match.
 		if (
 			phase === 'hit' ||
 			isUrl(this.path) ||
-			this.headers.normal.has('location')
+			this.headers.normal.has('location') ||
+			!!this.body
```

Comment:
so if `this.body` if defined here we always want to bail out, is that correct?

do you think that this can have unintended consequences? (i.e. can we get a truthy `this.body` and not want to bail out?)

Reply:
That is correct. There should be no unintended consequences. The value for `this.body` is only ever set when a middleware returns a response that is supposed to be returned to the client.

Reply:
awesome thanks 🙂